File: common/src/main/java/me/ftvkyo/fuzzy_broccoli/common/mvc/model/parts/Creature.java
package me.ftvkyo.fuzzy_broccoli.common.mvc.model.parts;

import org.joml.Vector2f;
import org.joml.Vector3f;


public class Creature {

    private Vector3f position;

    private Vector2f view;


    public Creature(Vector3f position, Vector2f view) {
        this.position = position;
        this.view = view;
    }


    public Vector3f getPosition() {
        return position;
    }


    public Vector2f getView() {
        return view;
    }
}

File: common/src/main/java/me/ftvkyo/fuzzy_broccoli/common/mvc/model/parts/Creature.kt
package me.ftvkyo.fuzzy_broccoli.common.mvc.model.parts

import org.joml.Vector2f
import org.joml.Vector3f


class Creature(val position: Vector3f, view: Vector2f) {
    var view = view
        set(value) {
            while (value.x > +180F) value.x += -360F
            while (value.x < -180F) value.x += +360F

            if (value.y > +90F) value.y = +90F
            if (value.y < -90F) value.y = -90F

            field = value
        }
}
